doc/modules/preprocessing.rst

```diff
@@ -474,11 +474,11 @@ then the shape of the transformed input is
     Imputer(add_indicator_features=True, axis=0, copy=True, missing_values='NaN',
         strategy='mean', verbose=0)
     >>> print(imp_with_in.transform(X))                           # doctest: +ELLIPSIS
-    [[ 4.          2.          1.          0.        ]
-     [ 6.          3.66666667  0.          1.        ]
+    [[ 4.          2.          0.          0.        ]
```

Comment:
Apologies!! This is really wrong. I have accidentally pushed the wrong commit from my other machine. I totally understand the API design and that the the fix is just about the shape of transformed output being consistent for different inputs. I will revert everything correctly. 

But one final question please - now indicator features will have a column for each feature regardless of that feature having missing values in the current transform input. But the imputed_features_ will have only features with at least one missing value. The reason is I can't do X[:, expected_imputer_features] because imputed_features won't have that feature but the indicator features will still have it. This is the only thing I suppose I didn't ask about. 

Sorry that I actually don't know what exactly how this indicator features help but I understand that they provide information about the missing data and thus can help improve the accuracy. And in case if you feel I am still not getting it right, please do let me know if it will be better for someone else to do this.